### PR TITLE
Histogram equalization for color images

### PR DESCRIPTION
**Pull request by Arcadie**
This changes the logic of the processing for color images when histogram equalization is selected:

- perform 98% contrast adjustment
- colorize image
- histogram equalization

Everything else should remain the same.

I'm converting the image to [Lab](https://en.wikipedia.org/wiki/CIELAB_color_space), equalizing the histogram for the L value [0..100], and then converting back to RGBA.

Here is the difference between telemetry / histogram with false color

<details>
  <summary>Large image warning: 6MB</summary>
  
telemetry-vs-historgam
  
</details>

### CODE REVIEW

**Comment by Arcadie** [comment]
I've finally managed to make the histogram equalization work "in-place", without copying channels around.
This is more in line with the "processing pipeline" without all the re-assignments (`let img = do_something(&img)`).

I should have marked this as _draft_ when I started, in case you started reviewing it and I still made some changes, sorry about that.

One thing to be aware of are the `unsafe` blocks in histogram equalization. It's the same algorithm used in `imageproc::contrast::equalize_histogram_mut` [src](https://docs.rs/imageproc/0.21.0/src/imageproc/contrast.rs.html#164-183), but adapted to work with sub-images, as accessing the pixels is done differently. 

Equalizing the histogram for the L channel happens in a similar way, but in range [0..100], so vector length is 101.

Please let me know if you have any suggestions or concerns, thanks!

File: src/noaa_apt.rs [hunk 1]
```diff
@@ -150,28 +151,41 @@ pub fn process(
...
     };
 
+    // for colorization & histogram equalization,
+    // always do 98% contrast adjust first, then colorize,
+    // then equalize histogram of color image if needed
+    if color.is_some() {
+        if let Contrast::Histogram = contrast_adjustment {
+            let (l, h) = misc::percent(&signal, 0.98)?;
+            low = l;
+            high = h;
```
Comment (martinber) [inline src/noaa_apt.rs:161]:
```suggestion
            (low, high) = misc::percent(&signal, 0.98)?;
```

**Review by martinber — APPROVED** [review]
It's OK the way it is, these are some suggestions that I will do if I find time

**Comment by martinber** [comment]
Very nice work. Sorry for taking so long. I will be very busy at least another week

File: src/noaa_apt.rs (hunk 1 above)
Comment (Arcadie) [reply src/noaa_apt.rs:160]: Unfortunately destructuring assignments are not yet supported: https://github.com/rust-lang/rfcs/issues/372 so this is a workaround.

**Comment by Arcadie** [comment]
> Sorry for taking so long. I will be very busy at least another week

No worries! I've addressed all the issues, thanks for taking the time to look at this!

**Comment by martinber** [comment]
Looks good!

**Pull request merged**